### PR TITLE
First bloc implementation

### PR DESCRIPTION
**Pull request by LeonardoMolleker**
ezgif com-gif-maker (1)
Implemented bloc logic to make the dices roll.

### CODE REVIEW

File: lib/bloc/dice_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  int getRandomNumber() {
+    return Random().nextInt(Constants.diceNumbers) + 1;
+  }
+
+  @override
+  Stream<List<int>> get stream => _streamController.stream;
+
+  @override
+  int get numberOfDices => diceNumbers;
+}
```
Comment (FranciscoGaraventa) [inline lib/bloc/dice_bloc.dart]: This method is unnecesary. Try to use the variable.

File: lib/bloc/dice_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import 'dart:async';
+import 'dart:math';
+
+import '../const/constants.dart';
+import 'i_dice_bloc.dart';
+
+class DiceBloc extends IDiceBloc {
+  int diceNumbers;
```
Comment (FranciscoGaraventa) [inline lib/bloc/dice_bloc.dart]: Change this name for a more representative like 'numberOfDice' or 'diceAmount' and do it final.

File: lib/bloc/dice_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  DiceBloc({
+    this.diceNumbers,
+  });
+
+  @override
+  Future<void> initialize() {}
+
+  List<int> defaultDices() {
+    List<int> defaultDices = List.generate(
+      numberOfDices,
```
Comment (FranciscoGaraventa) [inline lib/bloc/dice_bloc.dart:24]: Use the variable define in DiceBloc constructor.

File: lib/bloc/dice_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  @override
+  rollDices() {
+    _values.clear();
+    for (var i = 0; i < diceNumbers; i++) {
+      _values.add(getRandomNumber());
+    }
+    _streamController.sink.add(_values);
+  }
```
Comment (FranciscoGaraventa) [inline lib/bloc/dice_bloc.dart]: You forgot a return type in this method.

Comment (FranciscoGaraventa) [reply lib/bloc/dice_bloc.dart]: getRandomNumber() is not neccessary, because you use only here.

File: lib/bloc/dice_bloc.dart [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+  int getRandomNumber() {
+    return Random().nextInt(Constants.diceNumbers) + 1;
+  }
```
Comment (FranciscoGaraventa) [inline lib/bloc/dice_bloc.dart]: Remove this function.

File: lib/bloc/i_dice_bloc.dart [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import 'bloc.dart';
+
+abstract class IDiceBloc extends Bloc {
+  int get numberOfDices;
+
+  Stream<List<int>> get stream;
+
+  rollDices();
```
Comment (FranciscoGaraventa) [inline lib/bloc/i_dice_bloc.dart]: Add return type.

File: lib/bloc/i_dice_bloc.dart [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import 'bloc.dart';
+
+abstract class IDiceBloc extends Bloc {
+  int get numberOfDices;
```
Comment (FranciscoGaraventa) [inline lib/bloc/i_dice_bloc.dart]: This function is not necessary.

File: lib/main.dart [hunk 8]
```diff
@@ -1,3 +1,6 @@
+import 'package:dice_roller/bloc/i_dice_bloc.dart';
```
Comment (FranciscoGaraventa) [inline lib/main.dart]: Use relative import here.

File: lib/main.dart [hunk 9]
```diff
@@ -6,7 +9,10 @@ void main() {
 }
 
 class MyApp extends StatelessWidget {
-  // This widget is the root of your application.
+  final IDiceBloc bloc = DiceBloc(diceNumbers: 6);
```
Comment (FranciscoGaraventa) [inline lib/main.dart]: You missed a trailing comma here, and the number 6 should be the default number on DiceBloc constructor.

File: lib/src/pages/dice_page.dart [hunk 10]
```diff
@@ -85,4 +87,10 @@ class DicePage extends StatelessWidget {
       ),
     );
   }
+
+  List<Widget> buildList(AsyncSnapshot snapshot) {
+    return snapshot.data.map<Widget>((value) {
+      return Dice(bloc, path, ext, value);
```
Comment (FranciscoGaraventa) [inline lib/src/pages/dice_page.dart]: You missed a trailing comma here.

File: lib/bloc/dice_bloc.dart (hunk 1 above)
Comment (LeonardoMolleker) [reply lib/bloc/dice_bloc.dart]: Done!

File: lib/bloc/dice_bloc.dart (hunk 2 above)
Comment (LeonardoMolleker) [reply lib/bloc/dice_bloc.dart]: Done!

File: lib/bloc/dice_bloc.dart (hunk 3 above)
Comment (LeonardoMolleker) [reply lib/bloc/dice_bloc.dart:24]: Done!

File: lib/bloc/dice_bloc.dart (hunk 4 above)
Comment (LeonardoMolleker) [reply lib/bloc/dice_bloc.dart]: Fixed

File: lib/bloc/dice_bloc.dart (hunk 5 above)
Comment (LeonardoMolleker) [reply lib/bloc/dice_bloc.dart]: Done.

File: lib/bloc/i_dice_bloc.dart (hunk 6 above)
Comment (LeonardoMolleker) [reply lib/bloc/i_dice_bloc.dart]: Fixed.

File: lib/bloc/i_dice_bloc.dart (hunk 7 above)
Comment (LeonardoMolleker) [reply lib/bloc/i_dice_bloc.dart]: Removed.

File: lib/main.dart (hunk 8 above)
Comment (LeonardoMolleker) [reply lib/main.dart]: Fixed.

File: lib/main.dart (hunk 9 above)
Comment (LeonardoMolleker) [reply lib/main.dart]: Added.

File: lib/src/pages/dice_page.dart (hunk 10 above)
Comment (LeonardoMolleker) [reply lib/src/pages/dice_page.dart]: Added.

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM!

**Comment by marianosalvetti** [comment]
looks good for me.

**Pull request merged**